crates/core/src/db/messages/write.rs

```diff
@@ -28,60 +67,66 @@ impl Operation {
             _ => Self::Insert,
         }
     }
+
+    pub fn decode<'a>(reader: &mut impl BufReader<'a>) -> Result<Self, DecodeError> {
+        let flags = reader.get_u8()?;
+        let op = (flags & 0b1000_0000) >> 7;
+
+        Ok(Self::from_u8(op))
+    }
+
+    pub fn encoded_len(&self) -> usize {
+        1
+    }
+
+    pub fn encode(&self, writer: &mut impl BufWriter) {
+        let mut flags = 0u8;
+        flags = if self.to_u8() != 0 { flags | 0b1000_0000 } else { flags };
+        writer.put_u8(flags);
+    }
+}
+
+/// Error context for [`Write::decode`].
+enum Context {
```

Comment:
It's slightly weird to me that this name is so generic and doesn't mention errors in any way. If that is very idiomatic, I'm fine with it though.

Reply:
Well I don't know what's idiomatic, it's just a private type to make it convenient to exploit `anyhow::Context` for providing, well, context. My idea is to add another layer of context with the byte offset in #496.

If you prefer `ErrorContext`, `ErrCtx` or whatever, I'm happy to paint it like this.